plugins/search-react/src/components/HighlightedSearchResultText/HighlightedSearchResultText.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(
+  () => ({
+    highlight: {},
+  }),
+  { name: 'BackstageHighlightedSearchResultText' },
+);
+
+/**
+ * @public
+ */
+export type HighlightedSearchResultTextProps = {
+  text: string;
+  preTag: string;
+  postTag: string;
+};
+
+/**
+ * @public
+ */
+export const HighlightedSearchResultText = ({
+  text,
+  preTag,
+  postTag,
+}: HighlightedSearchResultTextProps) => {
+  const classes = useStyles();
+  const terms = text.split(new RegExp(`(${preTag}.+?${postTag})`));
+  return (
+    <>
+      {terms.map((t, idx) =>
+        t.includes(preTag) ? (
+          <mark className={classes.highlight} key={idx}>
+            {t.replace(new RegExp(`${preTag}|${postTag}`, 'g'), '')}
+          </mark>
+        ) : (
+          t
+        ),
+      )}
+    </>
+  );
```

Comment:
Good candidate for a quick test.  If there's a reasonable way to test the `BackstageHighlightedSearchResultText`-based style override, that'd be cool too?  But more of a nice-to-have.

Reply:
hmm not sure if a test for overriding the styling is very applicable here since that would more about testing the functionality of MUI style overrides rather than this component